### ISSUE TITLE
OpenTelemetry operation spans don't properly nest child spans (CosmosDb) [BUG]

### ISSUE THREAD

**Issue by andrejpk** [Azure/azure-sdk-for-java#28556]
**Describe the bug**
When OpenTelemetry is enabled, the CosmosDb creates Spans for calls inbound into the library (e.g. queryItems.[collection]) but instrumented options downstream (e.g. HTTP client calls) are not parented to the inbound library call span. This makes it harder to troubleshoot issues because it's not clear if that downstream operation was triggered by the CosmosDb client or something else later in the application code. 

***Exception or Stack Trace***
No exception/crash

**To Reproduce**
Configure:
- An OpenTelemetry collector (Jaeger, App Insights, etc.)
- Enable the CosmosDb OTel instrumentation
- If needed, enable auto-instrumentation on the appropriate http client (Netty, Apache HttpClient) that the CosmosDb client uses (I don't have to do this because the App Insights agent set all this up)

Make a query call into the CosmosDb SDK.

See the spans generated in the OpenTelemetry collector. Instead of the HTTP call to Cosmos being parented to the queryItems span, it is parented to the one of the code calling into the library.

***Code Snippet***

**Expected behavior**
Expect the CosmosDb client library to create and activate its span, so that any spans created during the work of that CosmosDb SDK operation are parented to it.

**Screenshots**
Here is an example showing how a CosmosDb library call appears in the App Insights e2e view vs another library that does this correctly (app code in this case):
![image](https://user-images.githubusercontent.com/1774478/165600296-9d4d3065-fbb2-4f44-9234-d9fe18306fce.png)

**Setup (please complete the following information):**
 - OS: MacOS
 - IDE: IntelliJ
 - Library/Libraries: [e.g. com.azure:azure-core:1.16.0 (groupId:artifactId:version)]
 - Java version: com.microsoft.azure:azure-cosmos:3.7.6
 - App Server/Environment: Spring Boot in AKS
 - Frameworks: Spring Boot using Spring Data via com.microsoft.azure:azure-cosmosdb-spring-boot-starter
 - App Insights agent: com.microsoft.azure:applicationinsights-agent:3.2.10

**Additional context**
This isn't app App Insights issue -- the behavior should be the same in any OTel collector

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
@lmolkova could you please take a look?

**Comment by lmolkova** [comment]
@andrejpk, thank you for reporting this.

Can you please help me clarify something? You mentioned you use **`com.microsoft.azure:azure-cosmos:3.7.6`, but it does not support distributed tracing**. You'd still see HTTP spans, but not `queryItems` span.

When I repro the problem with `com.microsoft.azure:azure-cosmos:3.7.6`, here's what I see - no Cosmos span:

![image](https://user-images.githubusercontent.com/2347409/169403734-8768fc13-28b8-42e7-a0f2-b62349aeba8c.png)


If I use the newer version, that supports distributed tracing `com.azure:azure-cosmos:4.29.1`

![image](https://user-images.githubusercontent.com/2347409/169403977-218266f4-1c24-4e77-a6d4-9cacf82add3b.png)

So, can you please confirm the version of the cosmos SDK you're using?

Also, we fixed a similar issue https://github.com/Azure/azure-sdk-for-java/pull/25571 and `com.azure:azure-cosmos:4.22.0` should have it. If you're using v4, can you please try updating it and see if the problem reproduces?

**Comment by andrejpk** [comment]
Hi @lmolkova, I unfortunately don't have access to that environment anymore since my work on that project wrapped up. I'll see what I can find though and let you know.

**Comment by andrejpk** [comment]
@lmolkova , I looked at this again and the Cosmos dependency comes through com.azure.spring:azure-spring-boot-starter-cosmos, but I'm not sure which version was in the build that produced that trace. (That project wasn't pinning versions.) It's possible this was on the 4.0 version at the time I captured these traces. That could explain what you said. If it's not relevant in 3.x and fixed in the 4.x versions, we can probably close this out.
